### PR TITLE
Fix adding empty Constant node after FormattedValue

### PR DESCRIPTION
**Pull request by lysnikolaou**
Before this PR, there was an empty `Constant` node after the `FormattedValue` node even for a simple f-string like `f'{a}'`:

```python
>>> f = 'f"{a}"'
>>> ast.dump(ast.parse(f))
"Module(body=[Expr(value=JoinedStr(values=[FormattedValue(value=Name(id='a', ctx=Load()), conversion=-1), Constant(value='')]))], type_ignores=[])"
```

With this PR:
```python
>>> f = 'f"{a}"'
>>> ast.dump(ast.parse(f))
"Module(body=[Expr(value=JoinedStr(values=[FormattedValue(value=Name(id='a', ctx=Load()), conversion=-1)]))], type_ignores=[])"
```

### CODE REVIEW

**Comment by pablogsal** [comment]
@lysnikolaou Does this need to wait until #149 is landed?

**Comment by lysnikolaou** [comment]
@pablogsal This can go already before #149, I think.

File: Parser/pegen.c
Comment (pablogsal) [inline Parser/pegen.c:2692]: I think we are leaking (and before as well) this. IIRC this needs to be added to the arena (`_PyArena_AddPyObject`) otherwise nobody is going to clean it. I may be missing something though

File: Parser/pegen.c
Comment (pablogsal) [inline Parser/pegen.c]:
```suggestion
    const char* _str = PyBytes_AsString(b->bytes);
```

File: Parser/pegen.c
Comment (lysnikolaou) [reply Parser/pegen.c:2692]: Yup, I think that's correct.

Comment (lysnikolaou) [reply Parser/pegen.c:2692]: Fixed in https://github.com/we-like-parsers/cpython/pull/164/commits/a257a531cff008931e9155092d3fc712761c99b8.

File: Parser/pegen.c
Comment (isidentical) [inline Parser/pegen.c]: Since you already added `str` to the arena, I don't think this is necessary.

Comment (lysnikolaou) [reply Parser/pegen.c]: Correct.

```suggestion
```

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
Yes!!

**Pull request merged**
